pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -824,4 +841,76 @@ public AuthenticationService getAuthenticationService() {
     public List<PersistentTopic> getAllTopicsFromNamespaceBundle(String namespace, String bundle) {
         return multiLayerTopicsMap.get(namespace).get(bundle).values();
     }
+    
+    public ZooKeeperDataCache<Map<String, String>> getDynamicConfigurationCache() {
+        return dynamicConfigurationCache;
+    }
+
+    /**
+     * Update dynamic-ServiceConfiguration with value present into zk-configuraiton-map and register listeners on
+     * dynamic-ServiceConfiguration field to take appropriate action on change of zk-configuraiton-map.
+     */
+    private void updateConfigurationAndRegisterListeners() {
+        // update ServiceConfiguration value by reading zk-configuration-map
+        updateDynamicServiceConfiguration();
+        // update brokerShutdownTimeoutMs value on listener notification
+        registerConfigurationListener("brokerShutdownTimeoutMs", new Consumer<String>() {
+            @Override
+            public void accept(String shutdownTime) {
+                pulsar.getConfiguration().setBrokerShutdownTimeoutMs(Integer.parseInt(shutdownTime));
+            }
+        });
+        //add more listeners here
+    }
+
+    /**
+     * Allows a listener to listen on update of {@link ServiceConfiguration} change, so listener can take appropriate
+     * action if any specific config-field value has been changed.
+     * </p>
+     * On notification, listener should first check if config value has been changed and after taking appropriate
+     * action, listener should update config value with new value if it has been changed (so, next time listener can
+     * compare values on configMap change).
+     * 
+     * @param configKey
+     *            : configuration field name
+     * @param listener
+     *            : listener which takes appropriate action on config-value change
+     */
+    public void registerConfigurationListener(String configKey, Consumer<String> listener) {
```

Comment:
Can we make the listener to be working for string, ints, longs? (Maybe leveraging the `FieldParser`)